### ISSUE TITLE
how to make IOT Agent to use HTTPS to communicate with Orion contextBroker?

### ISSUE THREAD

**Issue by guillemtnezh** [telefonicaid/iotagent-json#584]
I've just configured Orion CB to work (ONLY) with HTTPS. But now i need IOT Agent to make the requests through https. 

In my scenario, my device PUTS info ocassionally, which i'm simulating with a cUrl.

At node_modules>iotagent-node-lib>lib>services>ngsi>entities-NGSI-v1.js, it fails making the sendUpdateValueNgsi1 function, right here in the request:

> `

                request(
                    options,
                    generateNGSIOperationHandler('update', entityName, typeInformation, token, options, callback)
                );
`
Specifically it catches the error when entering at generateNGSIOperationHandler function. 
In my opinion it's because I'm doing an http request. Any idea of how to make that request in https?

thanks in advance

( @fgalan )

**Comment by fgalan** [comment]
NGSIv1 is an obsolete API at CB, soon to be removed (https://github.com/telefonicaid/iotagent-json/pull/533 https://github.com/telefonicaid/iotagent-node-lib/pull/995).

Could you configure the IOTA to work with NGSIv2 (check https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/installationguide.md for `contextBroker.ngsiVersion` / IOTA_CB_NGSI_VERSION configuration) and check if the problem persists, please?

**Comment by guillemtnezh** [comment]
Done, but the problem persists at the same analogous point.

I've realised of this during the starting of IOTA:

> `time=2021-07-26T18:09:41.348Z | lvl=DEBUG | corr=50064c45-8e4d-44db-a946-de8a836910b4 | trans=50064c45-8e4d-44db-a946-de8a836910b4 | op=IoTAgentNGSI.NorthboundServer | from=n/a | srv=n/a | subsrv=n/a | msg=Using config:
{
    "logLevel": "DEBUG",
    "timestamp": true,
    "contextBroker": {
        "host": "localhost",
        "port": "1026",
        "ngsiVersion": "v2",
        "fallbackTenant": "iotagent",
        "fallbackPath": "/",
        "url": "http://localhost:1026"
    },
    "server": {
        "port": 4041
    },
    "deviceRegistry": {
        "type": "mongodb"
    },
    "mongodb": {
        "host": "localhost",
        "port": "27017",
        "db": "iotagentjson"
    },
    "types": {},
    "service": "howtoService",
    "subservice": "/howto",
    "providerUrl": "http://localhost:4041",
    "deviceRegistrationDuration": "P20Y",
    "defaultType": "Thing",
    "defaultResource": "/iot/json",
    "explicitAttrs": false,
    "iotaVersion": "1.18.0-next",
    "multiCore": false,
    "relaxTemplateValidation": false,
    "defaultEntityNameConjunction": ":",
    "defaultExpressionLanguage": "legacy"
}
 | comp=IoTAgent`

I think maybe i should change 'url' to https://... but I don´t really know how

**Comment by guillemtnezh** [comment]
all right, just added directly to CB in config.js.

then I made a curl to update a value `curl -iX POST   'http://localhost:7896/iot/json?k=4jggokgpepnvsb2uv4s40d59ov2&i=motion002'   -H 'Content-Type: application/json'   -d '{"c": "4"}'` and received an Error of Self signed Certificate.
It`s true, I have  a self signed cert at the CB.

I modified curl adding the option '-k, --insecure      Allow insecure server connections when using SSL': but this only works for connection {CURL to IOTA} and not for {IOTA to CB}.

i'll keep working on it

**Comment by guillemtnezh** [comment]
Finally got to the point that i need either:
- iota to run insecure, which idk 
- put the cert into my root ca, which i cant manage to do it

**Comment by guillemtnezh** [comment]
I made it, but sadly I'm not sure exactly how, because i tried so many things.

The only reasonable thing i think i made was copying the cert that i use in the start `contextBroker -https -cert (...).pem -key (...).key` (got it from here https://github.com/telefonicaid/fiware-orion/blob/master/test/functionalTest/httpsPrepare.sh) into /usr/local/share/ca-certificates [changing name to finish with .crt] and then making an update-ca-certificates.

after that, it didnt work, but i tried to restart the computer and also orion. I killed the previous processes and started it again, but unconsciously in port 10026. It was working so i configured IOTa to ask there, and shows up that it finally works HTTPS between iota and CB.

sorry if I'm not explicit at all, that's all i can say.
